java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -0,0 +1,268 @@
+  public String constructor = "";
+
+  @RuleProperty(
+    key = "factoryMethod",
+    description = "the fully-qualified name of a factory method that returns an open resource, with or without a parameter list."
+      + " E.G. \"org.assoc.res.ResourceFactory#create\" or \"org.assoc.res.SpecialResourceFactory #create(java.lang.String, int)\"")
+  public String factoryMethod = "";
+
+  @RuleProperty(
+    key = "openingMethod",
+    description = "the fully-qualified name of a method that opens an existing resource, with or without a parameter list."
+      + " E.G. \"org.assoc.res.ResourceFactory#create\" or \"org.assoc.res.SpecialResourceFactory #create(java.lang.String, int)\"")
+  public String openingMethod = "";
+
+  @RuleProperty(
+    key = "closingMethod",
+    description = "the fully-qualified name of the method which closes the open resource, with or without a parameter list."
+      + " E.G. \"org.assoc.res.MyResource#closeMe\" or \"org.assoc.res.MySpecialResource#closeMe(java.lang.String, int)\"")
+  public String closingMethod = "";
+
+  private MethodInvocationMatcherCollection classConstructor;
+
+  private MethodInvocationMatcherCollection factoryList;
+  private MethodInvocationMatcherCollection openingList;
+  private MethodInvocationMatcherCollection closingList;
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    AbstractStatementVisitor visitor = new PreStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    PostStatementVisitor visitor = new PostStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  @Override
+  public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager constraintManager) {
+    List<ObjectConstraint> constraints = context.getState().getFieldConstraints(Status.OPENED);
+    for (ObjectConstraint constraint : constraints) {
+      Tree syntaxNode = constraint.syntaxNode();
+      String name = null;
+      if (syntaxNode.is(Tree.Kind.NEW_CLASS)) {
+        name = ((NewClassTree) syntaxNode).identifier().symbolType().name();
+      } else if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+        name = ((MethodInvocationTree) syntaxNode).symbolType().name();
+      }
+      if (name != null) {
+        context.reportIssue(syntaxNode, this, "Close this \"" + name + "\".");
+      }
+    }
+  }
+
+  private MethodInvocationMatcherCollection constructorClasses() {
+    if (classConstructor == null) {
+      classConstructor = MethodInvocationMatcherCollection.create();
+      if (constructor.length() > 0) {
+        classConstructor.add(MethodMatcherFactory.constructorMatcher(constructor));
+      }
+    }
+    return classConstructor;
+  }
+
+  private MethodInvocationMatcherCollection factoryMethods() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move this method into "PostStatementVisitor". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3398)